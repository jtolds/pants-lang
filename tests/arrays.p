output_list = { |x|
  i = 0
  while { < i x.size() } {
    print i ":" x[i]
    i := + i 1
  }

  print.
}

test_list = { |x|
  output_list x

  i = - 1
  while { >= i (- x.size()) } {
    print i ":" x[i]
    i := - i 1
  }

  print.

  try {
    print x[- x.size()]
    print x[- (- x.size()) 1]
  } { |e|
    print "out of range:" e
  }

  print.

  try {
    print x[- x.size() 1]
    print x[x.size()]
  } { |e|
    print "out of range:" e
  }

  print.
}

test_list ["a", "b", "c"]
test_list ["d", "e"]

v = ["hello", "mr", "sam"]
output_list v
v[2] = "fox"
output_list v
v[2] = "poopypants"
output_list v

x = []
x.append(3)
x.append(4)
x.unshift(5)
output_list(x)

print x.pop()
print.
output_list(x)
print x.shift()
print.
output_list(x)
