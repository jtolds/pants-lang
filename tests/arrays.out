0 : a
1 : b
2 : c

-1 : c
-2 : b
-3 : a

a
out of range: array index out of bounds!

c
out of range: array index out of bounds!

0 : d
1 : e

-1 : e
-2 : d

d
out of range: array index out of bounds!

e
out of range: array index out of bounds!

0 : hello
1 : mr
2 : sam

0 : hello
1 : mr
2 : fox

0 : hello
1 : mr
2 : poopypants


return code: 0
